### DISCUSSION TITLE
Directory Ownership in Container: root vs mambauser

### DISCUSSION THREAD

**Discussion by navneethc** [mamba-org/micromamba-docker#382] [Q&A] [answered]
We're attempting to move one of our images from using miniconda3 as the base image to micromamba (1.5-bookworm-slim).

Our repository contains a Makefile that has commands to build the image and run a few tests.

The build is nothing special, and so is the Dockerfile. (We briefly switch to `root` to install a package using APT.)

The test command is as follows:
```
docker run --mount type=bind,source="$$(pwd)/test-data",target=/assets --mount type=bind,source="$$(pwd)/test-output",target="/tmp/test-output"  image_name:tag [call to script and arguments omitted]
```
This eventually fails on my Ubuntu system because the script is not able to write to the /tmp/test-output directory because it is owned by `root`. However when a colleague runs it on his Mac, the owner of the folder is `mambauser` and the test runs successfully.

When I explicitly add `--user mambauser` to the `run` command, the owner shows up as `1000`, but the test still fails.

Short of setting the user to `root` in the Dockerfile, how can I get this to work? 

Happy to share relevant details of Dockerfile, system settings etc., if required.

**Comment by wholtz** [comment]
Hello @navneethc. I believe what you are seeing is due to a difference in the file permissions on `$(pwd)/test-output` on the host system.

On the Ubuntu system without using `docker`, are you able to create a file within `/tmp/test-output`?

**Reply by navneethc** [reply]
Hi @wholtz , thanks for the reply. Yes, I can create both the `test-output` directory and a file within it. Both are owned by the my user. `/tmp` itself is owned by root, of course. The `test-output` directory in the host system that I'm trying to mount is within the code repo.

```
~
❯ mkdir /tmp/test-output

~
❯ touch /tmp/test-output/foo.txt

~
❯ ls -l /tmp/test-output/
total 0
-rw-rw-r-- 1 <username> <username> 0 Nov  7 13:02 foo.txt

~
❯
```

**Reply by wholtz** [reply, marked as answer]
Great. Can you try `docker run -u "$(id -u):$(id -g)" ...` ?

**Reply by navneethc** [reply]
This worked to the satisfaction of everyone involved. Thank you very much for your help.
